release_dashboard/test/fakes/services/fake_conductor.dart

```diff
@@ -0,0 +1,46 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:conductor_core/src/proto/conductor_state.pb.dart';
+import 'package:conductor_ui/services/conductor.dart';
+
+import '../fake_start_context.dart';
+
+class FakeConductor extends ConductorService {
+  FakeConductor({
+    this.fakeStartContextProvided,
+  });
+
+  final FakeStartContext? fakeStartContextProvided;
+
+  @override
+  Future<void> createRelease({
+    required String candidateBranch,
+    required String? dartRevision,
+    required List<String> engineCherrypickRevisions,
+    required String engineMirror,
+    required List<String> frameworkCherrypickRevisions,
+    required String frameworkMirror,
+    required String incrementLetter,
+    required String releaseChannel,
+  }) async {
+    if (fakeStartContextProvided != null) {
+      return fakeStartContextProvided!.run();
+    } else {
+      /// If there is no [fakeStartContextProvided], initialize a fakeStartContext
+      /// with no exception thrown when [run] is called.
+      final FakeStartContext fakeStartContext = FakeStartContext();
+      return fakeStartContext.run();
+    }
+  }
+
+  @override
+  ConductorState? get state {
```

Comment:
Ok, I changed it to
```
@override
ConductorState? state = ConductorState(
  conductorVersion: 'abcdef',
  releaseChannel: 'dev',
  releaseVersion: '2.7.0-4.0.pre',
);
```
How do we change it in tests in the future?

Reply:
```
// This callback function can be attached to the handler of a button or some other trigger that will happen in your test
void Function() callback = () => fakeConductor.state = updatedState;

tester.tap(someButtonThatCallsCallback);
// assert that the state was really updated
expect(fakeConductor.state.interestingField == updatedState.interestingField, true);
```

Reply:
Adding this right now...

Reply:
Regarding this issue, could we do something like [this](https://github.com/Yugue/cocoon/blob/cdaeeca77c8907701e3970a54b5427d1cf666277/release_dashboard/test/fakes/services/fake_conductor.dart#L54) for `fake_conductor.dart`? And have `testState` as a parameter in `FakeConductor` [here](https://github.com/Yugue/cocoon/blob/cdaeeca77c8907701e3970a54b5427d1cf666277/release_dashboard/test/fakes/services/fake_conductor.dart#L17)? 

In that way, it might be easier to pass a testState to `FakeConductor` by doing [this](https://github.com/Yugue/cocoon/blob/cdaeeca77c8907701e3970a54b5427d1cf666277/release_dashboard/test/widgets/conductor_status_test.dart#L111). Otherwise, three lines is required in order to pass a testState by doing this:
```
    FakeConductor fakeConductor = FakeConductor();
    fakeConductor.state = state;
    await tester.pumpWidget(MyApp(fakeConductor));
    ```